### DISCUSSION TITLE
Migration having multiple databases

### DISCUSSION THREAD

**Discussion by julianCast** [phalcon/cphalcon#15602] [Q&A]
My config.php where I keep my databases in an array 'databses'
```
return new Phalcon\Config([
  'databases' => [
    'db' => [
      'adapter' => 'Postgresql',
      'host' => $environment_conf['POSTGRESQL_HOST'],
      'port' => $environment_conf['POSTGRESQL_PORT'],
      ...
    ],
    'db_stats' => [
      'adapter' => 'Postgresql',
      'host' => $environment_conf['POSTGRESQL_HOST_STATS'],
      'port' => $environment_conf['POSTGRESQL_PORT_STATS'],
      ...
    ]
  ],
```
My index.php
```
  foreach ($config->databases as $db => $cred) {
    $di->set($db, function () use ($cred) {
      $adaptor = "\\Phalcon\\Db\\Adapter\\Pdo\\{$cred->adapter}";
      return new $adaptor([
        'host' => $cred->host,
        'username' => $cred->username,
        'password' => $cred->password,
         ...
      ]);
    });
  }
```

I have no problem with controllers because I use setConnectionService to change to the desired db in the modelBase controller.
```
public function onConstruct()
  {
    $this->setConnectionService('db_stats');
    $this->setSchema('user_usage');
  }
```

But for migration I can't find how to indicate that my config does not have 'database' parameter, but an array instead.

`phalcon migration run --migrations=migrations`
<img width="343" alt="Screenshot 2021-07-26 at 16 40 00" src="https://user-images.githubusercontent.com/559677/127113892-162ea971-b611-4763-9136-ea8fb2b3f249.png">

The only solution that I can think of is having one config.php file per each DB.
Thank you!

**Comment by Jeckerson** [comment]
Hello. You are using very old devtools, if you can use Phalcon v4, please look at this migrations config
https://github.com/phalcon/migrations#create-configuration-file

**Reply by julianCast** [reply]
You are right but as far as I know, the problem is that the command `phalcon migration run`  will look, not matter what, into the property 'database' inside the config, making imposible to keep an array of databases in the config.php file.

<img width="943" alt="Screenshot 2021-07-27 at 13 44 55" src="https://user-images.githubusercontent.com/559677/127476348-c02ccaa8-4336-4045-828c-8c970626eeb3.png">
